dtls.c

```diff
@@ -684,12 +684,37 @@ static void dtls_debug_keyblock(dtls_security_parameters_t *config)
 		  dtls_kb_iv_size(config, peer->role));
 }
 
+static size_t
+print_byte(char *buf, size_t buflen, uint8_t value) {
+  char const *bufstart = buf;
+  uint8_t quot = 100;
+
+  if (value == 0) {
+    *buf++ = '0';
+    buflen--;
+  } else {
+    /* adjust quot to avoid leading zeroes */
+    while (quot && (value / quot == 0))
+      quot /= 10;
+
+    while (quot && buflen) {
+      *buf++ = '0' + (value / quot);
+      buflen--;
+      value %= quot;
+      quot /= 10;
+    }
+  }
+  return buf - bufstart;
+}
+
 /** returns the name of the goven handshake type number.
   * see IANA for a full list of types:
   * https://www.iana.org/assignments/tls-parameters/tls-parameters.xml#tls-parameters-7
   */
 static const char *dtls_handshake_type_to_name(int type)
 {
+ static char buf[16] = "unknown (";;
```

Comment:
Code as:

```
#ifdef DTLS_CONSTRAINED_STACK
  dtls_mutex_lock(&static_mutex);
#endif /* DTLS_CONSTRAINED_STACK */
```

makes me wonder, if concurrent processing is assumed?

Reply:
Interesting question - multi-threading is not supported as far as I am aware.  Looks like I added this in some time back to remove data held on the stack into a static area if the stack was constrained in size and wanted to make sure things were properly protected.